src/tools/interop/idt/idt.py

```diff
@@ -0,0 +1,237 @@
+        discover_parser.set_defaults(func=self.command_discover)
+        discover_parser.add_argument(
+            "--type",
+            "-t",
+            help="Specify the type of discovery to execute",
+            required=True,
+            choices=[
+                "ble",
+                "b",
+                "mdns",
+                "m"])
+
+        capture_parser = subparsers.add_parser(
+            "capture",
+            help="Capture all information of interest while running a manual test")
+
+        platform_help = "Run capture for a particular platform"
+        if self.available_platforms_default:
+            platform_help += f" (default {self.available_platforms_default})"
+        capture_parser.add_argument("--platform",
+                                    "-p",
+                                    help=platform_help,
+                                    required=self.platform_required,
+                                    choices=self.available_platforms,
+                                    default=self.available_platforms_default)
+
+        # TODO: Ability for each ecosystem to specify its own custom arguments
+        capture_parser.add_argument(
+            "--ecosystem",
+            "-e",
+            help="Run capture for a particular ecosystem or ALL ecosystems (default ALL)",
+            required=False,
+            choices=self.available_ecosystems,
+            default=self.available_ecosystems_default)
+
+        capture_parser.add_argument("--pcap",
+                                    "-c",
+                                    help="Run packet capture (default t)",
+                                    required=False,
+                                    choices=['t', 'f'],
+                                    default='t')
+
+        interface_help = "Run packet capture against a specified interface"
+        if self.available_net_interfaces_default:
+            interface_help += f" (default {self.available_net_interfaces_default})"
+        capture_parser.add_argument(
+            "--interface",
+            "-i",
+            help=interface_help,
+            required=self.net_interface_required,
+            choices=self.available_net_interfaces,
+            default=self.available_net_interfaces_default)
+
+        # TODO: Implement
+        capture_parser.add_argument(
+            "--additional",
+            "-a",
+            help="Run ble and mdns scanners in the background while capturing (default t)",
+            required=False,
+            choices=[
+                't',
+                'f'],
+            default='t')
+
+        capture_parser.set_defaults(func=self.command_capture)
+
+        args, unknown = parser.parse_known_args()
+        if not hasattr(args, 'func'):
+            parser.print_help()
+        else:
+            args.func(args)
+
+    def command_discover(self, args: argparse.Namespace) -> None:
+        if args.type[0] == "b":
```

Comment:
I feel the tool should constantly, passively monitor on all available channels. It shouldn't provide the operator various modes.

Reply:
As discover is meant to be interactive,each option requires its own shell, and the command is split accordingly.